rust/src/types.rs

```diff
@@ -938,6 +938,128 @@ impl Type {
         let mut name = QualifiedName::from(name);
         unsafe { BnStr::from_raw(BNGenerateAutoDemangledTypeId(&mut name.0)) }
     }
+
+    pub fn demangle_gnu3<Mangled: BnStrCompatible>(
+        arch: &CoreArchitecture,
+        mangled_name: Mangled,
+        simplify: bool,
+    ) -> Result<(Option<Type>, Vec<String>)> {
+        let mangled_name_bwn = mangled_name.as_bytes_with_nul();
+        let mangled_name_ptr = mangled_name_bwn.as_ref();
+        unsafe {
+            let mut out_type: *mut BNType = std::mem::zeroed();
+            let mut out_name: *mut *mut std::os::raw::c_char = std::mem::zeroed();
+            let mut out_size: usize = 0;
+            let mut names = Vec::new();
+            let res = BNDemangleGNU3(
+                arch.0,
+                mangled_name_ptr.as_ptr() as *const i8,
+                &mut out_type,
+                &mut out_name,
+                &mut out_size,
+                simplify,
+            );
+
+            if !res || out_size == 0 {
+                let cstr = match CStr::from_bytes_with_nul(mangled_name_ptr) {
+                    Ok(cstr) => cstr,
+                    Err(_) => {
+                        log::error!("demangle_gnu3: failed to parse mangled name");
+                        return Err(());
+                    }
+                };
+                return Ok((None, vec![cstr.to_string_lossy().into_owned()]));
+            }
+
+            let out_type = match out_type.is_null() {
+                true => {
+                    log::debug!("demangle_gnu3: out_type is NULL");
+                    None
+                }
+                false => Some(Type::from_raw(out_type)),
+            };
+
+            if out_name.is_null() {
+                log::error!("v: out_name is NULL");
```

Comment:
v?

Reply:
That was me trying to paste and not pressing the command key seemingly... Thank you, sorry for not noticing!